test/utils.py

```diff
@@ -100,14 +101,64 @@ class TestFrameInfo:
     duration_seconds: float
 
 
+@dataclass
+class TestVideoStreamInfo:
+    width: int
+    height: int
+    num_color_channels: int
+
+
+@dataclass
+class TestAudioStreamInfo:
+    sample_rate: int
+    num_channels: int
+    duration_seconds: float
+
+
 @dataclass
 class TestContainerFile:
     filename: str
 
-    # {stream_index -> {frame_index -> frame_info}}
+    default_stream_index: int
+    stream_infos: Dict[int, Union[TestVideoStreamInfo, TestAudioStreamInfo]]
     frames: Dict[int, Dict[int, TestFrameInfo]]
 
-    default_stream_index: int
+    def __post_init__(self):
+        # We load the .frames attribute from the checked-in json files, if needed.
+        # These frame info files are dumped with ffprobe, e.g.:
+        # ```
+        # ffprobe -v error -hide_banner -select_streams v:1 -show_frames -of json test/resources/nasa_13013.mp4 | jq '[.frames[] | {duration_time, pts_time}]'
+        # ```
+        # This will output the metadata for the frames of the second video
+        # stream (v:1). First audio stream would be a:0.
+        # Note that we are using the absolute stream index in the file. But
+        # ffprobe uses a relative stream for that media type.
+        for stream_index in self.stream_infos:
+            if stream_index in self.frames:
+                # .frames may be manually set: for some streams, we don't need
+                # the info for all frames. We don't need to load anything in
+                # this case
+                continue
```

Comment:
I don't feel too strongly about this but if we were to enforce the existence of a json file for every single asset, it would raise the barrier to the first test when e.g. adding a new test asset, which I think isn't desirable.